### PR TITLE
Added a changelog feature

### PR DESCRIPTION
**Pull request by robojumper**
The changelog feature crawls through the steam website and gets the changelog, with html tags filtered out.

### CODE REVIEW

**Comment by robojumper** [comment]
Nah, i somehow borked the export feature. Please double-check what went wrong.

File: xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs [hunk 1]
```diff
@@ -138,5 +141,27 @@ public override string ToString()
         {
             return ID;
         }
+
+        public void GetChangelog(RichTextBox modinfo_changelog_richtextbox)
```
Comment (BlueRaja) [inline xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs:145]: The method signature should be `string GetChangelog()`.  It shouldn't rely on `RichTextBox`

File: xcom2-launcher/xcom2-launcher/Forms/MainForm.cs [hunk 2]
```diff
@@ -392,6 +393,10 @@ private void UpdateExport()
         }
         #endregion
 
+        private void richTextBox1_TextChanged(object sender, EventArgs e)
```
Comment (BlueRaja) [inline xcom2-launcher/xcom2-launcher/Forms/MainForm.cs:396]: This doesn't seem to do anything?  It should be removed.

File: xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs (hunk 1 above)
Comment (robojumper) [inline xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs:145]: The Webclient is a sync though

**Comment by BlueRaja** [comment]
Most mods do not even use the changelog, and a user may have a lot of mods, so making a web request for every mod every time `UpdateModInfo()` is called is a **huge** overhead.

Instead it should:
- Only make the web-request when the user actually switches to the 'changelog' tab
- Cache the request so the changelog is only requested once per mod.

The lifetime of this application is short-lived, so there's probably no reason to ever update the cache.

File: xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs (hunk 1 above)
Comment (BlueRaja) [inline xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs:145]: So make the method `async` as well.

**Comment by robojumper** [comment]
Gonna close the pull request and rework the code as well as figure out why the export feature broke.

**Pull request closed without merging**